test/nodes/test_label_generator.py

```diff
@@ -48,15 +62,20 @@ def test_pseudo_label_generator_using_question_document_pairs(
     ]
     psg = PseudoLabelGenerator(docs, retriever)
     train_examples = []
-    for idx, doc in enumerate(document_store):
-        # the documents passed here are ignored as we provided source documents in the constructor
-        output, stream = psg.run(documents=[doc])
+    if batch:
+        output, pipe_id = psg.run_batch(documents=document_store.get_all_documents())
         assert "gpl_labels" in output
         for item in output["gpl_labels"]:
             assert "question" in item and "pos_doc" in item and "neg_doc" in item and "score" in item
             train_examples.append(item)
 
-    assert len(train_examples) > 0
+    else:
+        output, pipe_id = psg.run(documents=document_store.get_all_documents())
+        assert "gpl_labels" in output
+        for item in output["gpl_labels"]:
+            assert "question" in item and "pos_doc" in item and "neg_doc" in item and "score" in item
+            train_examples.append(item)
 
+    assert len(train_examples) > 0
     retriever.train(train_examples)
     retriever.save(tmp_path)
```

Comment:
I understand this is not part of the diff, but why we're running these two calls if we don't check their results? By the way, we could test these in an another unit tests? :grin:  (same for the previous test)

Reply:
Yep I would test these two in a separate test that does the following (more or less):

- Sets up everything
- Calls retriever.train()
- Calls retriever.save()
- Checks that the files are saved properly
- [If doable] Checks that the saved files are any different from the original ones (to ensure the training did something). You could use a local model to start from, so you have something to compare.

We would still be checking two things at once, but given that train is hardly testable in isolation, I think it's alright to do it here :+1: